package/MDAnalysis/analysis/hbond_analysis.py

```diff
@@ -0,0 +1,204 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""Hydrogen Bond Analysis --- :mod:`MDAnalysis.analysis.hbond_analysis`
+===========================================================================
+
+:Author: Paul Smith and Mateusz Bieniek
+:Year: 2019
+:Copyright: GNU Public License v3
+
+This module provides methods to find and analyse hydrogen bonds in a Universe.
+
+The :class:`HydrogenBondAnalysis` class is a new version of the original :class:`HydrogenBondAnalysis`
+class from the module mod:`MDAnalysis.analysis.hbonds.hbond_analysis`, which itself was modeled after the `VMD
+HBONDS plugin`_.
+
+.. _`VMD HBONDS plugin`: http://www.ks.uiuc.edu/Research/vmd/plugins/hbonds/
+
+
+Options:
+  - *d_h_cutoff* (Å) [1.2] : Distance cutoff used for finding donor-hydrogen pairs
+  - *d_a_cutoff* (Å) [3.0] : Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance.
+  - *d_h_a_angle_cutoff* (degrees) [150] : D-H-A angle cutoff for hydrogen bonds.
+
+
+Output:
+  - *frame* : frame at which a hydrogen bond was found
+  - *donor id* : atom id of the hydrogen bond donor atom
+  - *hydrogen id* : atom id of the hydrogen bond hydrogen atom
+  - *acceptor id* : atom id of the hydrogen bond acceptor atom
+  - *distance* (Å): length of the hydrogen bond
+  - *angle* (degrees): angle of the hydrogen bond
+
+Hydrogen bond data are returned in a :class:`numpy.ndarray` on a "one line, one observation" basis
+and can be accessed via :attr:`HydrogenBondAnalysis.hbonds`::
+
+    results = [
+        [
+            <frame>,
+            <donor index (0-based)>,
+            <hydrogen index (0-based)>,
+            <acceptor index (0-based)>,
+            <distance>,
+            <angle>
+        ],
+        ...
+    ]
+
+Classes
+-------
+
+.. autoclass:: HydrogenBondAnalysis
+   :members:
+
+   .. attribute:: hbonds
+
+"""
+from __future__ import absolute_import
+
+import numpy as  np
+
+from . import base
+from MDAnalysis.lib.distances import capped_distance, calc_angles
+
+class HydrogenBondAnalysis(base.AnalysisBase):
+    """
+    Perform an analysis of hydrogen bonds in a Universe.
+    """
+
+    def __init__(self, universe, donors_sel, hydrogens_sel, acceptors_sel, d_h_cutoff, d_a_cutoff, d_h_a_angle_cutoff):
+        """Set up atom selections and geometric criteria for finding hydrogen bonds in a Universe.
+
+        Parameters
+        ----------
+        universe : Universe
+            MDAnalysis Universe object
+        donors_sel : str
+            Selection string for the hydrogen bond donor atoms
```

Comment:
Providing atom groups would definitely be nice, it's something to think about.

Reply:
This could be neatly solved with Universe.selectAtoms(AG) returning AG. This way all interfaces would automatically accept str and AG selections.

Reply:
I thought the same. Raise a separate issue. It looks like a simple change but this is such a fundamental method that it will require some careful testing. If it works consistently then it would be great. I think there was an old issue related to the "use AG where strings are allowed". 

There are some cases where it won't just work, in particular anywhere where strings are used to create "updating AtomGroups" by executing the same selection every frame (I think the waterdyamics module does this and probably elsewhere, too). Anyway, separate issue!

Reply:
@richardjgowers can you raise an issue for your idea that `select_atoms` should be able to pass through ags, please?

Reply:
raised #2369